AltairZ80/altairz80_cpu.c

```diff
@@ -7359,6 +7369,9 @@ static t_stat cpu_hex_load(FILE *fileref, CONST char *cptr, CONST char *fnam, in
 
         bufptr = datastr;
 
+        /* Ensure datastr is not NULL */
+        datastr[strlen(datastr)] = '\0';

```

Comment:
That doesn't do anything.  "strlen" returns the offset at which a 0 byte is found, so if strlen works then this just writes 0 over that existing 0.  And if there isn't a 0 termination in place, strlen isn't valid.

Reply:
Thanks, @pkoning2 for pointing that out.  I will upload a new patch set to resolve.